src/query/src/optimizer.rs

```diff
@@ -12,6 +12,28 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use datafusion_common::config::ConfigOptions;
+use datafusion_common::Result;
+use datafusion_expr::LogicalPlan;
+
+use crate::QueryEngineContext;
+
+/// [`ExtensionAnalyzerRule`]s transform [`LogicalPlan`]s in some way to make
+/// the plan valid prior to the rest of the DataFusion optimization process.
+/// It's an extension of datafusion [`AnalyzerRule`]s but accepts [`QueryEngineContext` as the second parameter.
+pub trait ExtensionAnalyzerRule {
```

Comment:
`AnalyzerRuleExtension` would be better. It's an extension to analyzer rules, not analyzer rules to extensions.

Reply:
Just keep the same as `DistExtensionPlanner`.

Other cases such as `substrait::extension_serializer::ExtensionSerializer`.